Mathlib/Topology/MetricSpace/Pseudo/Defs.lean

```diff
@@ -401,6 +401,14 @@ theorem sphere_eq_empty_of_subsingleton [Subsingleton α] (hε : ε ≠ 0) : sph
 instance sphere_isEmpty_of_subsingleton [Subsingleton α] [NeZero ε] : IsEmpty (sphere x ε) := by
   rw [sphere_eq_empty_of_subsingleton (NeZero.ne ε)]; infer_instance
 
+theorem closedBall_eq_singleton_of_unique [Unique α] (h : 0 ≤ ε) : closedBall x ε = {x} := by
```

Comment:
Instead of assuming `Unique`, could you assume `Subsingleton` and `Nonempty`? The difference is that `Unique` requires data (a `default` element should be given) so it can sometimes lead to nontrivial business of lack of definitional equality, while the two typeclasses `Subsingleton` and `Nonempty` are Prop-valued and avoid this issue. Unless a theorem about `Unique` uses `default`, it's always better to spell it with `Subsingleton` and `Nonempty`.

Reply:
:bulb: Thanks for enlightening *me*!

Reply:
(And there's an in-progress linter about this.)